hledger-lib/Hledger/Data/Journal.hs

```diff
@@ -752,8 +753,7 @@ journalInferCommodityStyles j =
 commodityStylesFromAmounts :: [Amount] -> M.Map CommoditySymbol AmountStyle
 commodityStylesFromAmounts amts = M.fromList commstyles
   where
-    samecomm = \a1 a2 -> acommodity a1 == acommodity a2
-    commamts = [(acommodity $ head as, as) | as <- groupBy samecomm $ sortBy (comparing acommodity) amts]
+    commamts = [(acommodity $ head as, as) | as <- groupSortBy (comparing acommodity) amts]
```

Comment:
Isn't this is the same case you shown earlier?

```haskell
commamts = groupSort [(acommodity as, as) <- amts]
```

Reply:
Indeed, thanks